test/suites/secondary_redis.bash

```diff
@@ -0,0 +1,96 @@
+SUITE_secondary_redis_PROBE() {
+    if ! command -v redis-server &> /dev/null; then
+        echo "redis-server not found"
+        return
+    fi
+    if ! command -v redis-cli &> /dev/null; then
+        echo "redis-cli not found"
+        return
+    fi
+}
+
+SUITE_secondary_redis_SETUP() {
+    unset CCACHE_NODIRECT
+    export CCACHE_SECONDARY_STORAGE="redis://localhost:7777"
+
+    generate_code 1 test.c
+}
+
+expect_number_of_cache_entries() {
+    local expected=$1
+    local host=$2
+    local port=$3
+    local actual
+
+    actual=$(redis-cli -h "$host" -p "$port" keys "ccache:*" | wc -l)
+    if [ "$actual" -ne "$expected" ]; then
+        test_failed_internal "Found $actual (expected $expected) entries in $host:$port"
+    fi
+}
+
+SUITE_secondary_redis() {
+    if $HOST_OS_APPLE; then
+         # no coreutils on darwin by default, perl rather than gtimeout
+         function timeout() { perl -e 'alarm shift; exec @ARGV' "$@"; }
+    fi
+    timeout 10 redis-server --bind localhost --port 7777 &
+    sleep 0.1 # wait for boot
+    redis-cli -p 7777 ping
+    # shellcheck disable=SC2064
+    trap "kill $!" EXIT
```

Comment:
You should be able to delete these two lines since the test framework now (as of a28ad9db64f39b49652bf2ad8ab8d3259786d33b) kills child processes started by a test suite.

Reply:
Sounds good. Maybe can add that when adding a test for the unix socket path variant (only on non-Windows)